tests/test_api.py

```diff
@@ -1,2 +1,74 @@
+"""
+Minimum possible tests for Object Store API
+
+Not got into detail of mocking components and responses -
+is there much value or do we end up testing FastAPI?
+
+At level of "do endpoints exist, and resolve"
+"""
+import os
+from os_api.api import app, s3_endpoint
+import pytest
+import aioboto3
+from fastapi import FastAPI
+from fastapi.testclient import TestClient
+from moto import mock_aws
+import logging
+logging.basicConfig(level=logging.INFO)
+
+from moto.server import ThreadedMotoServer
+
+@pytest.fixture(scope="module")
+def moto_server():
+    """Fixture to run a mocked AWS server for testing."""
+    # Note: pass `port=0` to get a random free port.
+    server = ThreadedMotoServer(port=0)
+    server.start()
+    host, port = server.get_host_and_port()
+    os.environ['AWS_URL_ENDPOINT'] = f"http://{host}:{port}"
+    yield f"http://{host}:{port}"
+    server.stop()
+
+# Without both the dependency override _and_ the environment variable set in the fixture,
+# the endpoint URL doesn't get set for the API properly - wish i fully understood why! - JW
+app.dependency_overrides[s3_endpoint] = moto_server
```

Comment:
It would take me a while to form a basic enough understanding of what's going on here to be able to lend a hand - sorry! For what it's worth I'm happy to move forward without fully understanding what's going on with a testing utility, but I'll try to keep this in mind.

Reply:
As probably mentioned in the PR this is very much a "hammer it until it starts working" line of code1

https://github.com/fastapi/fastapi/blob/9416e89bd7e3491157f8c962012aebd7be7ed5d6/docs_src/dependency_testing/tutorial001.py#L32 - this is a more legible example of simpler usage. 

https://github.com/ocaballeror/pytest-aioboto3 - i read through this to try and improve understanding, but it doesn't apply neatly in this case where we are testing the API code but only have direct access to the response to the client during the tests (which is what causes the need for `dependency_override` rather than a more direct [`mock` of code you are calling](https://docs.python.org/3/library/unittest.mock.html) )

I tend to try and avoid complex mocks in tests because you can spend ages scratching your head over them, only to discover that you've bypassed all your own code and are only testing that third party dependencies work as expected. I quite like this `moto_server` way of doing it (spin up a temporary interface just for your process that responds like s3 does, grab its address and set that in the environment where your code is running) even if it feels a bit kludgy. Definitely less effort and cost than [running `localstack` in docker](https://github.com/NERC-CEH/rse_group/discussions/37) though!